src/public/class-wordlift-context-cards.php

```diff
@@ -28,81 +30,126 @@ function __construct() {
 				}
 			} else {
 				$response['description'] = $jsonld[0]['description'];
 			}
 		}
 
-		if( isset($jsonld[0]['name']) && !empty($jsonld[0]['name']) ){
+		if ( isset( $jsonld[0]['name'] ) && ! empty( $jsonld[0]['name'] ) ) {
 			$response['title'] = $jsonld[0]['name'];
 		}
 
-		if( isset($jsonld[0]['url']) && !empty($jsonld[0]['url']) ){
+		if ( isset( $jsonld[0]['url'] ) && ! empty( $jsonld[0]['url'] ) ) {
 			$response['url'] = $jsonld[0]['url'];
 		}
 
-		if( isset($jsonld[0]['image']) &&
-		    isset($jsonld[0]['image'][0]['url']) &&
-		    isset($jsonld[0]['image'][0]['width']) &&
-		    isset($jsonld[0]['image'][0]['height'])
-		){
+		if ( isset( $jsonld[0]['image'] ) &&
+		     isset( $jsonld[0]['image'][0]['url'] ) &&
+		     isset( $jsonld[0]['image'][0]['width'] ) &&
+		     isset( $jsonld[0]['image'][0]['height'] )
+		) {
 			$response['image'] = array(
-				'url' => $jsonld[0]['image'][0]['url'],
-				'width' => $jsonld[0]['image'][0]['width'],
+				'url'    => $jsonld[0]['image'][0]['url'],
+				'width'  => $jsonld[0]['image'][0]['width'],
 				'height' => $jsonld[0]['image'][0]['height'],
 			);
 		}
 
-		if($publisher){
-			$publisher_id       = Wordlift_Configuration_Service::get_instance()->get_publisher_id();
-			$publisher_jsonld   = Wordlift_Jsonld_Service::get_instance()->get_jsonld( false, $publisher_id );
+		if ( $publisher ) {
+			$publisher_id          = Wordlift_Configuration_Service::get_instance()->get_publisher_id();
+			$publisher_jsonld      = Wordlift_Jsonld_Service::get_instance()->get_jsonld( false, $publisher_id );
 			$response['publisher'] = $this->format_response( $publisher_jsonld, false );
 		}
 
 		return $response;
 	}
 
+	function format_as_jsonld( $jsonld, $publisher = true ) {
+
+		$response = array();
+
+		if ( ! isset( $jsonld ) || empty( $jsonld ) || empty( $jsonld[0] ) ) {
+			return null;
+		}
+
+		$publisher_id     = Wordlift_Configuration_Service::get_instance()->get_publisher_id();
+		$publisher_jsonld = Wordlift_Jsonld_Service::get_instance()->get_jsonld( false, $publisher_id );
+
+		$response['@context']  = "http://schema.org";
+		$response['@type']     = "WebSite";
+		$response['publisher'] = $publisher_jsonld;
+		$response['mentions']  = $jsonld;
+
+		return $response;
+
+	}
+
 	public function context_data( $request ) {
 
 		$entity_uri = urldecode( $request->get_param( 'entity_url' ) );
-		$entity_id  = $this->url_to_postid( $entity_uri );
+		$entity_id  = Wordlift_Context_Cards_Service::url_to_postid( $entity_uri );
 		$jsonld     = Wordlift_Jsonld_Service::get_instance()->get_jsonld( false, $entity_id );
 
 		return $this->format_response( $jsonld );
 
 	}
 
+	public function context_data_jsonld( $request ) {
+
+		$ids = $request->get_param( 'id' );
+
+		// Look for an entity.
+		foreach ( $ids as $id ) {
+			$post = Wordlift_Entity_Service::get_instance()
```

Comment:
@akshayraje this can fire many heavy queries to the database, use the `Wordlift_Cached_Entity_Uri_Service`:

```
$cached_entity_uri_service->preload_uris( $uris );

foreach ( $ids as $id ) {
    $cached_entity_uri_service->get_entity( $uri );

   ...
```